File: app11_swift_ios/SwiftGuardData/Sources/SwiftGuardData/Models/Enums.swift
// PLAN.md §5.1 — core enums shared across the SwiftData model.

public enum Severity: String, Codable, Sendable {
    case critical, high, medium, low, info
}

public enum StrideCategory: String, Codable, Sendable {
    case s, t, r, i, d, e
}

public enum SampleType: String, Codable, Sendable {
    case attackDemo, defense
}

public enum CodeLanguage: String, Codable, Sendable, CaseIterable {
    case python, java, go, scala, lua
}

public enum MitigationType: String, Codable, Sendable {
    case preventive, detective, corrective, compensating
}

public enum Effort: String, Codable, Sendable {
    case low, medium, high
}

public enum Effectiveness: String, Codable, Sendable {
    case partial, significant, full
}

public enum RelationshipType: String, Codable, Sendable {
    case equivalent, related, parentChild, mapsTo
}

public enum AppLocale: String, Codable, Sendable {
    case polish = "pl"
    case english = "en"

    public var bundleIdentifier: String {
        rawValue
    }
}

File: app12_kotlin_android/data/src/main/kotlin/com/kotlinguard/data/model/Enums.kt
package com.kotlinguard.data.model

// PLAN.md §5.1 — core enums shared across the Room entity model.

enum class Severity { CRITICAL, HIGH, MEDIUM, LOW, INFO }
enum class StrideCategory { S, T, R, I, D, E }
enum class SampleType { ATTACK_DEMO, DEFENSE }
enum class CodeLanguage { PYTHON, JAVA, GO, SCALA, LUA }
enum class MitigationType { PREVENTIVE, DETECTIVE, CORRECTIVE, COMPENSATING }
enum class Effort { LOW, MEDIUM, HIGH }
enum class Effectiveness { PARTIAL, SIGNIFICANT, FULL }
enum class RelationshipType { EQUIVALENT, RELATED, PARENT_CHILD, MAPS_TO }

enum class AppLocale(val code: String) {
    POLISH("pl"),
    ENGLISH("en");

    companion object {
        fun fromCode(code: String): AppLocale? = entries.find { it.code == code }
    }
}
